telemetry/statsd/src/main/scala/io/buoyant/telemetry/statsd/Metric.scala

```diff
@@ -0,0 +1,31 @@
+package io.buoyant.telemetry.statsd
+
+import com.timgroup.statsd.StatsDClient
+import com.twitter.finagle.stats.{Counter => FCounter, Stat => FStat}
+import java.util.concurrent.atomic.AtomicInteger
+
+private[statsd] object Metric {
+
+  // counters batch up deltas and flush on send
+  class Counter(statsDClient: StatsDClient, name: String) extends FCounter {
+    val count = new AtomicInteger(0)
+
+    def incr(delta: Int): Unit = {
+      val _ = count.addAndGet(delta)
+    }
+
+    def send: Unit = statsDClient.count(name, count.getAndSet(0))
```

Comment:
fwiw i'm considering rolling this particular change back in favor of simple counter sampling.

initially i liked the idea of getting full-fidelity counters via batching, but counter sampling seems to be the statsd way. it will also will make this code simpler and avoid adding state to linkerd.